### PR TITLE
Functional Tests - Add BO tests for Brands

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Adding Functional tests for BO Brands and addresses
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | URL_FO=SHOP_URL TEST_PATH="functional/BO/catalog/brandsAndSuppliers/brands/*" npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16125)

### CODE REVIEW

File: tests/puppeteer/pages/BO/brands.js [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+      this.page.click(this.bulkActionsToggleButton.replace('%TABLE', 'manufacturer')),
+      this.page.waitForSelector(
+        `${this.bulkActionsToggleButton}[aria-expanded='true']`.replace('%TABLE', 'manufacturer'),
+        {visible: true},
+      ),
+    ]);
+    // Click on delete and wait for modal
+    await Promise.all([
+      this.page.click(enable ? this.bulkActionsEnableButton : this.bulkActionsDisableButton),
+      this.page.waitForNavigation({waitUntil: 'networkidle0'}),
```
Comment (nesrineabdmouleh) [inline tests/puppeteer/pages/BO/brands.js]: Please use the function `clickAndWaitForNavigation`

File: tests/puppeteer/campaigns/data/faker/brandAddress.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const faker = require('faker');
+
+const countries = ['Italy', 'France', 'United States', 'Netherlands', 'United Kingdom', 'Germany'];
+
+module.exports = class branAddress {
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/data/faker/brandAddress.js]:
```suggestion
module.exports = class brandAddress {
```

File: tests/puppeteer/campaigns/functional/BO/catalog/brandsAndSuppliers/brands/01_filterAndQuickEditBrands.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    browser = await helper.createBrowser();
+    page = await helper.newTab(browser);
+    this.pageObjects = await init();
+  });
+  after(async () => {
+    await helper.closeBrowser(browser);
+  });
+  // Login into BO and go to brands page
+  loginCommon.loginBO();
+  it('should go to brands page', async function () {
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/functional/BO/catalog/brandsAndSuppliers/brands/01_filterAndQuickEditBrands.js]: Maybe start to comment all the `it()` like we said we should ?

File: tests/puppeteer/campaigns/utils/globals.js [hunk 4]
```diff
@@ -18,7 +18,7 @@ global.INSTALL = {
 global.BROWSER_CONFIG = {
   headless: JSON.parse(process.env.HEADLESS || true),
   timeout: 0,
-  slowMo: 25,
+  slowMo: 5,
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/utils/globals.js]: 👍

File: tests/puppeteer/pages/BO/addBrand.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    return this.getTextContent(this.alertSuccessBlockParagraph);
+  }
+
+  /**
+   * Delete all keywords
+   * @param lang, to specify which input to empty
+   * @return {Promise<void>}
+   */
+  async deleteKeywords(lang = 'en') {
+    const closeButtons = await this.page.$$(this.deleteKeywordLink.replace('%LANG', lang));
```
Comment (SimonGrn) [inline tests/puppeteer/pages/BO/addBrand.js:86]: Might be a good idea to `trim()` and `.toLowerCase()` the `lang` parameter ?
Just to be sure

File: tests/puppeteer/pages/BO/addBrand.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  }
+
+  /**
+   * Add keywords
+   * @param keywords, array of keywords
+   * @param id, to choose which lang (1 for en, 2 for fr)
+   * @return {Promise<void>}
+   */
+  async addKeywords(keywords, id = '1') {
+    /* eslint-disable no-await-in-loop, no-restricted-syntax */
```
Comment (SimonGrn) [inline tests/puppeteer/pages/BO/addBrand.js:101]: This is becoming a problem: it's cluttering the code for no particular good reason.
Is it possible to put it in the `.eslint.rc` file ?

File: tests/puppeteer/pages/BO/brands.js [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+  }
+
+  /**
+   * Delete Row in table
+   * @param table, brand or address
+   * @param row, row to delete
+   * @return {Promise<textContent>}
+   */
+  async deleteRowInTable(table, row = '1') {
+    this.dialogListener(true);
```
Comment (SimonGrn) [inline tests/puppeteer/pages/BO/brands.js:212]: What is this for ?

Comment (boubkerbribri) [reply tests/puppeteer/pages/BO/brands.js:212]: To accept javascript dialog with message "Delete Selected items"
image

File: tests/puppeteer/pages/BO/addBrand.js (hunk 6 above)
Comment (boubkerbribri) [reply tests/puppeteer/pages/BO/addBrand.js:101]: Yes, but maybe in a new refacto :smile:

File: tests/puppeteer/pages/BO/addBrand.js (hunk 5 above)
Comment (boubkerbribri) [reply tests/puppeteer/pages/BO/addBrand.js:86]: we don't have to tolerate incorrect strings for lang, 
Lang is used by a selector and test should fail if language is wrong.

File: tests/puppeteer/campaigns/utils/globals.js (hunk 4 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/utils/globals.js]: I delete that modification for sanity tests. it's too fast and tests fails

Comment (SimonGrn) [reply tests/puppeteer/campaigns/utils/globals.js]: Oh damn. That's a problem...

File: tests/puppeteer/pages/BO/addBrand.js (hunk 5 above)
Comment (SimonGrn) [reply tests/puppeteer/pages/BO/addBrand.js:86]: That's a good reason indeed.

File: tests/puppeteer/campaigns/data/faker/brandAddress.js (hunk 2 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/data/faker/brandAddress.js]: done thanks

File: tests/puppeteer/pages/BO/brands.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/puppeteer/pages/BO/brands.js]: done thanks

File: tests/puppeteer/campaigns/functional/BO/catalog/brandsAndSuppliers/brands/01_filterAndQuickEditBrands.js (hunk 3 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/functional/BO/catalog/brandsAndSuppliers/brands/01_filterAndQuickEditBrands.js]: :+1:

File: tests/puppeteer/campaigns/utils/globals.js (hunk 4 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/utils/globals.js]: I will try something else but in a different PR

**Review by SimonGrn — DISMISSED** [review]

**Review by matthieu-rolland — APPROVED** [review]
lgtm

**Pull request merged**